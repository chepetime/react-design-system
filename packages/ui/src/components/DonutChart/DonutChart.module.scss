@use "./../../styles" as styles;

:root {
  --donutchart-background-color: #{styles.$pink-500};
}

:global .dark {
  --donutchart-background-color: #{styles.$indigo-500};
}

.DonutChart {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--donutchart-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
