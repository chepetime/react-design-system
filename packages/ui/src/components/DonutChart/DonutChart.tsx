import * as React from "react";
import classnames from "classnames";
import css from "./DonutChart.module.scss";

export interface DonutChartProps {
  children?: React.ReactNode;
}

export const DonutChart = React.forwardRef<HTMLDivElement, DonutChartProps>(
  (props: DonutChartProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.DonutChart)} ref={ref}>
        <p>DonutChart component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

DonutChart.displayName = "DonutChart";
