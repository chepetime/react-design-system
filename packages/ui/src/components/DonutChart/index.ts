export { DonutChart } from "./DonutChart";
export type { DonutChartProps } from "./DonutChart";