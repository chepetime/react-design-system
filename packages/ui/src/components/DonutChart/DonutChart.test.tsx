
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { DonutChart } from "./DonutChart";
import { render, screen } from "@testing-library/react";

describe("DonutChart", () => {
  it("renders without errors", () => {
    render(<DonutChart />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<DonutChart>Some children</DonutChart>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<DonutChart>Some children</DonutChart>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<DonutChart />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
