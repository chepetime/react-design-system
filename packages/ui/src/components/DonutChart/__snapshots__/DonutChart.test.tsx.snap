// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart matches snapshot 1`] = `
<div>
  <div
    class="DonutChart"
  >
    <p>
      DonutChart component working!
    </p>
    <div />
  </div>
</div>
`;
